synapse/storage/databases/main/deviceinbox.py

```diff
@@ -153,12 +156,21 @@ def get_new_messages_for_device_txn(txn):
             txn.execute(
                 sql, (user_id, device_id, last_stream_id, current_stream_id, limit)
             )
+
             messages = []
+            stream_pos = current_stream_id
```

Comment:
Why is this added? In case `txn` is empty?

Reply:
It's mostly just to stop my editor complaining that `stream_pos` is never set, though we should never get into a state where `stream_pos` is `None`. We'll either pull something from the database, or we won't have and then `len(messages) < limit`...

Well, unless `limit` is 0, but that'd be unlikely to be set.

So yes, just a peace of mind safeguard really. I can add a comment above it explaining what we're doing with `stream_pos` overall.